.github/workflows/user_yaml_ci.yaml

```diff
@@ -20,10 +20,21 @@ jobs:
     name: Test and deploy
     runs-on: ubuntu-latest
     steps:
-    - name: Checkout repository
-      uses: actions/checkout@v3
-      with:
-        persist-credentials: false
+    - name: Check PR
+      run: |
+        echo Event name: ${{ github.event_name }}
+        if [[ ${{ github.event_name }} = "pull_request" || ${{ github.event_name }} = "pull_request_target" ]]; then
+          is_fork=`curl --header "Authorization: token ${{ secrets.GITHUB_TOKEN }}" https://api.github.com/repos/${{ github.event.repository.full_name }}/pulls/${{github.event.number}} | jq -r .head.repo.fork`
```

Comment:
using the github API to get fork status and labels because the github workflow environment variables that are supposed to contain this information are unreliable